File: frontend/src/games/tichu/dtos/TichuDto.ts
import { TichuRule } from "@/games/tichu/domain/TichuRule.ts";
import { Team } from "@/games/tichu/domain/Team.ts";
import { CardRank } from "@/games/tichu/domain/Card.ts";
import { PhaseStatus, RoundStatus } from "@/games/tichu/domain/TichuGame.ts";
import { TichuDeclaration } from "@/games/tichu/domain/TichuDeclaration.ts";
import { PlayerIndex } from "@/games/tichu/types.ts";
import { CardDto } from "@/games/tichu/dtos/CardDto.ts";
import { TrickDto } from "@/games/tichu/dtos/TrickDto.ts";

export interface TichuDto {
  rule: TichuRule;
  players: PlayerDto[];
  scoresHistory: number[][];
  handCounts: Record<number, number>;
  myHand: CardDto[];
  roundStatus: RoundStatus;
  myExchange: { left: CardDto | null, mid: CardDto | null, right: CardDto | null };
  tichuDeclarations: (TichuDeclaration | null)[];
  wish: CardRank | null;
  exitOrder: (0 | 1 | 2)[];
  phaseStatus: PhaseStatus | null;
  turn: PlayerIndex | null;
  tricks: TrickDto[] | null;
}

export interface PlayerDto {
  id: number;
  name: string;
  team: Team;
}

File: src/main/java/com/icube/sim/tichu/games/tichu/dtos/TichuDto.java
package com.icube.sim.tichu.games.tichu.dtos;

import com.icube.sim.tichu.games.tichu.PhaseStatus;
import com.icube.sim.tichu.games.tichu.RoundStatus;
import com.icube.sim.tichu.games.tichu.TichuDeclaration;
import com.icube.sim.tichu.games.tichu.TichuRule;
import lombok.Builder;
import lombok.Getter;
import org.jspecify.annotations.Nullable;

import java.util.List;
import java.util.Map;

@Builder
@Getter
public class TichuDto {
    private final TichuRule rule;
    private final List<PlayerDto> players;
    private final List<int[]> scoresHistory;
    private final Map<Long, Integer> handCounts;
    private final List<CardDto> myHand;
    private final RoundStatus roundStatus;
    private final TichuDeclaration[] tichuDeclarations;
    private final @Nullable ExchangeSend myExchange;
    private final @Nullable Integer wish;
    private final int[] exitOrder;
    private final @Nullable PhaseStatus phaseStatus;
    private final @Nullable Integer turn;
    private final @Nullable List<TrickDto> tricks;
}
